### DISCUSSION TITLE
Redirect back to dapp after connect/pair app with metamask wallet

### DISCUSSION THREAD

**Discussion by Sabriel1984** [WalletConnect/walletconnect-monorepo#5144] [Developer Support]
I have Android and iOS apps with web3modal library (in kotlin multiplatform)

I have latest versions of wallet-connect bom and metamask.

Describe the bug
redirect back to Android dapp stops working in 7.24

Expected behavior
after successful pair dapp with metamask it should go back to original dapp - it worked properly at least on Android version

Steps to reproduce
use dapp with wallet connect to show web3modal dialog
choose metamask from wallets list
metamask will be opened - sign in if necessary
confirm that you want to connect dapp with metamask
and after connect success (I can see toast message in metamask) it should redirect user to original dapp

**Comment by glitch-txs** [comment]
Hello, this is an issue on the MetaMask side you should contact them instead
